docs/concepts.md

```diff
@@ -0,0 +1,123 @@
+    <pl-question-panel>
+        If $x = {{params.x}}$ and $y$ is {{params.operation}} $x$, what is $y$?
+    </pl-question-panel>
+
+    <pl-number-input answers-name="y" label="$y =$"></pl-number-input>
+    <pl-submission-panel> {{feedback.y}} </pl-submission-panel>
+    ```
+
+=== "Example `server.py`"
+
+    ```python title="server.py"
+    def generate(data):
+        data["params"]["x"] = random.randint(5, 10)
+        data["params"]["operation"] = random.choice(["double", "triple"])
+
+        if data["params"]["operation"] == "double":
+            data["correct_answers"]["y"] = 2 * data["params"]["x"]
+        else:
+            data["correct_answers"]["y"] = 3 * data["params"]["x"]
+
+    def parse(data):
+        if "y" not in data["format_errors"] and data["submitted_answers"]["y"] < 0:
+            data["format_errors"]["y"] = "Negative numbers are not allowed"
+
+    def grade(data):
+        if math.isclose(data["score"], 0.0) and data["submitted_answers"]["y"] > data["params"]["x"]:
+            data["partial_scores"]["y"]["score"] = 0.5
+            data["score"] = 0.5
+            data["feedback"]["y"] = "Your value for $y$ is larger than $x$, but incorrect."
+    ```
+
+=== "Example `info.json`"
+
+    ```json title="info.json"
+    {
+        "uuid": "cbf5cbf2-6458-4f13-a418-aa4d2b1093ff",
+        "title": "Doubling and Tripling",
+        "topic": "Basic Arithmetic",
+        "tags": ["sp24"],
+        "type": "v3",
+    }
+    ```
+
+### Elements
+
+Questions are written in HTML and **Elements**. Elements are pre-built interface elements for common questions, like numerical inputs, multiple choice, and more. **Custom Elements** are reusable components that can be tailored to your course and provide a way to create more complex questions. Custom elements are not necessary for most courses.
+
+=== "Integer input element"
+
+    ![](elements/pl-integer-input.png)
+
+=== "HTML for element"
+
+    ```html title="question.html"
+    <pl-integer-input answers-name="int_value" label="$y =$"></pl-integer-input>
+    ```
+
+You can view a list of all the available elements in the [Elements documentation](./elements.md).
+
+### Assessments
+
+A course contains a collection of questions, which are composed together to create [**Assessments**](./assessment/index.md). Each course instance contains a collection of assessments. There are **Exam Assessments** and **Homework Assessments**.
+
+Asssessments are organized into **Assessment Sets** based on the type of assessment (e.g. `Homework`, `Quiz`, `Exam`). Optionally, each assessment can be a part of a **Module** (e.g. `Introduction`, `Review`, `Linked Lists`) that represent a segment of a course. Your assessments can then be shown to students grouped by module or by assessment set.
+
+=== "Assessments (Module Grouping)"
+
+    ![alt text](img/module-grouping.png)
+
+=== "Assessments (Set Grouping)"
+
+    ![alt text](img/set-grouping.png)
+
+### Metadata
```

Comment:
We don't really use this word elsewhere to refer to these files, though I agree that's more or less what they are. Maybe call this section "JSON configuration files".